### PR TITLE
Added Lab 3 and README file

### PR DESCRIPTION
**Pull request by NoFaceBoy**
The code from the third lab finds your lucky number. First of all, you need to enter N number (max 10 numbers), after that choose your K key (1-9). Code forms new number M from every K symbol. When the number is formed it should be multiple of three, only then it is considered lucky. Program prints down that number and tells you, YES or NO depending on if the number is lucky or not.
Example:
N = 123456789
K = 3
M = 369485271
YES

### CODE REVIEW

File: lab3.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+void main() {
+	LuckyNumber();
+}
+void LuckyNumber() {
+	long n = 0;
+	int k = 0;
+	printf("Enter your number (max 10 digits): ");
+	scanf_s("%i", &n);
+	printf("Enter your key (1-9): ");
+	scanf_s("%i", &k);
```
Comment (RybOlya) [inline lab3.c:11]: The function executes too many functions at once, i recommend getting number from user in main and pass it to function

File: lab3.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#include <stdio.h>
+
+void LuckyNumber();
+void main() {
+	LuckyNumber();
+}
+void LuckyNumber() {
+	long n = 0;
```
Comment (RybOlya) [inline lab3.c]: Is it necessary to use long?

File: lab3.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	int k = 0;
+	printf("Enter your number (max 10 digits): ");
+	scanf_s("%i", &n);
+	printf("Enter your key (1-9): ");
+	scanf_s("%i", &k);
+	int num = 0;
```
Comment (RybOlya) [inline lab3.c]: Your n and num variables are similar, rename one

File: lab3.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	int* ptrReversed = (int*)malloc(10 * sizeof(int));
+	
+	while (n != 0)
+	{
+		ptrReversed[num] = n % 10;
+		n = n / 10;
+		num++;
+	}
+	int z = 0;
```
Comment (RybOlya) [inline lab3.c]: In for cycles, variable i is more used

File: lab3.c [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+			sum += ptrM[j];
+			j++;
+		}
+		else
+			i++;
+	} while (newnum != 0);
+	for (i = 0; i < num; i++) {
+		printf("%i", ptrM[i]);
+	}
+	if (sum % 3 == 0)
```
Comment (RybOlya) [inline lab3.c:69]: I recommend returning value from function to main print  out YES or NO according to the return value in main

File: lab3.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	{
+		ptrA[z] = ptrReversed[num - 1 - z];
+	}
+	int i = -1;
+	int newnum = num;
+	int j = 0;
+	int* ptrM = (int*)malloc(10 * sizeof(int));
+	int sum = 0;
+	do {
+		int f = k;
```
Comment (RybOlya) [inline lab3.c]: What is the meaning of f variable?

File: lab3.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		{
+			i++;
+			if (i >= num) {
+				i -= num;
+			}
+			if (ptrA[i] != 10) {
+				f--;
+			}
+		} while (f != 0);
+		if (ptrA[i] != 10) {
```
Comment (RybOlya) [inline lab3.c]: Avoid negative conditionals

File: lab3.c [hunk 8]
```diff
@@ -0,0 +1,64 @@
+#include <stdio.h>
```
Comment (RybOlya) [inline lab3.c:1]: Add comments to your code

**Review by RybOlya — CHANGES_REQUESTED** [review]

File: lab3.c (hunk 1 above)
Comment (NoFaceBoy) [reply lab3.c:11]: Fixed

File: lab3.c (hunk 2 above)
Comment (NoFaceBoy) [reply lab3.c]: Changed

File: lab3.c (hunk 3 above)
Comment (NoFaceBoy) [reply lab3.c]: Renamed

File: lab3.c (hunk 4 above)
Comment (NoFaceBoy) [reply lab3.c]: Changed

File: lab3.c (hunk 5 above)
Comment (NoFaceBoy) [reply lab3.c:69]: Fixed

File: lab3.c (hunk 6 above)
Comment (NoFaceBoy) [reply lab3.c]: k variable is used to save the key value, f variable used to move through available numbers, changed the names for them

File: lab3.c (hunk 7 above)
Comment (NoFaceBoy) [reply lab3.c]: I marked used numbers as 10, so I won't get them to the new number again, that's why it is used like that

File: lab3.c (hunk 8 above)
Comment (NoFaceBoy) [reply lab3.c:1]: added

**Review by RybOlya — APPROVED** [review]

File: lab3.c [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+char* LuckyNumber();
+void main() 
+{
+	int n;
+	int k;
```
Comment (zveres-ss) [inline lab3.c:6]: do not use single-letter variable names

File: lab3.c [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+	printf("Enter your key (1-9): ");
+	scanf_s("%i", &k);
+;
+	printf("%s", LuckyNumber(n,k));
+}
+
+char* LuckyNumber(int n, int k) {
+	int arrnum = 0;
+	int reversed[10];
```
Comment (zveres-ss) [inline lab3.c:18]: make 10 as constant or even better to use #define EXPLAINABLE_NAME 10

**Review by zveres-ss — CHANGES_REQUESTED** [review]

**Pull request merged**